src/sentry/static/sentry/app/views/discover/index.tsx

```diff
@@ -101,24 +100,26 @@ class DiscoverContainer extends React.Component<Props, State> {
     }
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    if (!nextProps.params.savedQueryId) {
-      this.setState({savedQuery: null});
-      // Reset querybuilder if we're switching from a saved query
-      if (this.props.params.savedQueryId) {
-        const {datetime, projects} = nextProps.selection;
+  componentDidUpdate(prevProps: Props, _prevState: State) {
+    const currProps = this.props;
+    const currState = this.state;
+
+    // Always ensure GlobalSelectionHeader is correct
+    // This doesn't seem too expensive to run at every update
+    this.setGlobalSelectionDate(currState.savedQuery);
```

Comment:
Why not check if the savedQuery has changed? (I assume that's the only scenario where we want to update things)